includes/mailpoet/class-sensei-mailpoet-repository.php

```diff
@@ -0,0 +1,120 @@
+<?php
+/**
+ * File containing the class Sensei_MailPoet_Repository.
+ *
+ * @since $$next-version$$
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Some Helper methods to fetch Sensei data: courses, groups and students.
+ */
+class Sensei_MailPoet_Repository {
+	/**
+	 * Get all enrolled students in a course or group.
+	 *
+	 * @param int    $id The post ID.
+	 * @param string $post_type The post type.
+	 *
+	 * @return array
+	 */
+	public static function get_students( $id, $post_type ) {
+		global $wpdb;
+		if ( 'group' === $post_type ) {
+			if ( class_exists( 'Sensei_Pro_Student_Groups\Repositories\Group_Student_Repository', true ) ) {
+				$group_student_repo = new Sensei_Pro_Student_Groups\Repositories\Group_Student_Repository( $wpdb );
+				$student_ids        = $group_student_repo->find_group_students( $id );
+			}
+		}
+		if ( 'course' === $post_type ) {
+			$students    = get_comments(
+				array(
+					'post_id' => $id,
+					'type'    => 'sensei_course_status',
+					'status'  => 'any',
+				)
+			);
+			$student_ids = array_map(
+				static function( $student ) {
+					return $student->user_id;
+				},
+				$students
+			);
+		}
+		if ( ! $student_ids ) {
+			return array();
+		}
+
+		$args = array(
+			'include' => $student_ids,
+			'fields'  => array( 'id', 'user_email', 'display_name', 'user_nicename' ),
+		);
+		return get_users( $args );
+	}
+
+	/**
+	 * Generates a Sensei LMS prefixed name for a MailPoet list.
+	 *
+	 * @param string $name The name of the course or group.
+	 *
+	 * @param string $post_type The post type: course or group.
+	 *
+	 * @return string
+	 */
+	public static function get_list_name( $name, $post_type ) {
+		return 'Sensei LMS ' . ucfirst( $post_type ) . ': ' . $name;
```

Comment:
Do you mean translating the 'Sensei LMS' part or the entire string that is returned?

Reply:
The safest probably is go with sprintf and a string with placeholders, like `__( 'Sensei LMS %s: %s', ...)`, in case a different ordering makes sense for some language.

Also maybe it could be enough to add 'Course: ' and 'Group: ' prefixes, without Sensei?